src/lib/badges/Badge.svelte

```diff
@@ -31,36 +29,70 @@
+    pink: 'bg-pink-100 text-pink-800 dark:bg-gray-700 dark:text-pink-400 border-pink-400'
+  };
+
+  const hoverColors = {
+    blue: 'hover:bg-blue-200',
+    dark: 'hover:bg-gray-200',
+    red: 'hover:bg-red-200',
+    green: 'hover:bg-green-200',
+    yellow: 'hover:bg-yellow-200',
+    indigo: 'hover:bg-indigo-200',
+    purple: 'hover:bg-purple-200',
+    pink: 'hover:bg-pink-200'
   };
 
   let badgeClass: string;
   $: badgeClass = classNames(
     baseClass,
-    large ? 'text-sm font-medium' : 'text-xs font-semibold',
-    colors[color],
-    href && (hovers[color] ?? hovers.blue),
-    rounded ? 'rounded-full p-1' : 'rounded px-2.5 py-0.5',
+    large ? 'text-sm' : 'text-xs',
+    bordered ? `border ${borderedColors[color]}` : colors[color],
+    href && hoverColors[color],
+    rounded ? 'rounded-full' : 'rounded',
     index && 'absolute font-bold border-2 border-white dark:border-gray-900',
     index && (large ? 'w-7 h-7 -top-3 -right-3' : 'w-6 h-6 -top-2 -right-2'),
     $$props.class
   );
+
+  const closeBtnBaseClass = 'inline-flex items-center p-0.5 ml-2 text-sm bg-transparent rounded-sm';
+
+  const closeBtnColors = {
+    blue: 'text-blue-400 hover:bg-blue-200 hover:text-blue-900 dark:hover:bg-blue-800 dark:hover:text-blue-300',
+    dark: 'text-gray-400 hover:bg-gray-200 hover:text-gray-900 dark:hover:bg-gray-600 dark:hover:text-gray-300',
+    red: 'text-red-400 hover:bg-red-200 hover:text-red-900 dark:hover:bg-red-800 dark:hover:text-red-300',
+    green: 'text-green-400 hover:bg-green-200 hover:text-green-900 dark:hover:bg-green-800 dark:hover:text-green-300',
+    yellow: 'text-yellow-400 hover:bg-yellow-200 hover:text-yellow-900 dark:hover:bg-yellow-800 dark:hover:text-yellow-300',
+    indigo: 'text-indigo-400 hover:bg-indigo-200 hover:text-indigo-900 dark:hover:bg-indigo-800 dark:hover:text-indigo-300',
+    purple: 'text-purple-400 hover:bg-purple-200 hover:text-purple-900 dark:hover:bg-purple-800 dark:hover:text-purple-300',
+    pink: 'text-pink-400 hover:bg-pink-200 hover:text-pink-900 dark:hover:bg-pink-800 dark:hover:text-pink-300'
+  };
+
+  let animation = false;
+  let hidden = false;
+  
   const handleHide = () => {
-    hidden = !hidden;
+    animation = true;
+    setTimeout(() => {
+      hidden = true;
+    }, 300);
   };
 </script>
 
-<svelte:element this={href ? 'a' : 'span'} {href} {...$$restProps} class={badgeClass} class:hidden>
+<svelte:element this={href ? 'a' : 'span'} {href} {...$$restProps} class={badgeClass} class:transition-opacity={animation} class:duration-300={animation} class:ease-out={animation} class:opacity-0={animation} class:hidden>
   <slot />
   {#if dismissable}
-    <CloseButton {color} on:click={handleHide} size={large ? 'sm' : 'xs'} class="ml-1.5 -mr-1.5" />
-  {/if}
+    <button on:click={handleHide} type="button" class={`${closeBtnBaseClass} ${closeBtnColors[color]}`} aria-label="Remove">
```

Comment:
As stated already, I'm not a fan of hardcoding the button here. I think it'd be better to use a slot:
```svelte
{#if dismissable}
<slot name="close_button">
  <CloseButton {color} on:click={handleHide} size={large ? 'sm' : 'xs'} class="ml-1.5 -mr-1.5" />
</slot>
{/if}
```